### PR TITLE
entity update

### LINKED ISSUE #2153: While riding horses, riding height is not adjusted to the size of the horse.

### PR DESCRIPTION
**Pull request by eerussianguy**
- Add leashable tag for wild animals that aren't normaly leashable. Adding animals to the tag allows using a lead on them
- Added plucking birds. This harms the bird and also adds wear.
- Buffed feather drop rates
- Added deer
- Added ability for predators/prey to be babies
- Added ability for any animal to spawn as a baby
- Refactored entity renderers
- Prevent animals from hurting themselves on bushes
- Baby livestock follow temptations more eagerly
- Predators are much better swimmers now
- Improved path navigation for entities to treat TFC water/leaves more intelligently
- nest box renders eggs
- Horses kick overburdened players off
- oviparous animals jump higher
- fix horse ride height (Closes #2153)
- rats
- rat infestations in chests

### CODE REVIEW

File: resources/constants.py [hunk 1]
```diff
@@ -922,18 +925,22 @@ def lang_enum(name: str, values: Sequence[str]) -> Dict[str, str]:
     'subtitles.entity.tfc.cougar.death': 'Cougar dies',
     'subtitles.entity.tfc.cougar.attack': 'Cougar roars',
     'subtitles.entity.tfc.cougar.ambient': 'Cougar screams',
-    'subtitles.entity.tfc.cougar.hurt': "Cougar yowls",
-    'subtitles.entity.tfc.cougar.sleep': "Cougar snores",
+    'subtitles.entity.tfc.cougar.hurt': 'ugar yowls',
```
Comment (alcatrazEscapee) [inline resources/constants.py]: Typo?

File: resources/data.py [hunk 2]
```diff
@@ -390,7 +393,7 @@ def plank(_variant: str) -> str:
     rm.block_tag('rabbit_raidable', 'tfc:crop/carrot', 'tfc:crop/cabbage', 'minecraft:carrots')
     rm.block_tag('single_block_replaceable', 'tfc:groundcover/humus', 'tfc:groundcover/dead_grass')
     rm.block_tag('powder_snow_replaceable', '#minecraft:dirt', '#forge:gravel', '#tfc:grass', 'minecraft:snow')
-    rm.item_tag('usable_on_tool_rack', 'tfc:firestarter', 'minecraft:bow', 'minecraft:crossbow', 'minecraft:flint_and_steel')
```
Comment (alcatrazEscapee) [inline resources/data.py:393]: Is this an unintentional deletion?

File: src/main/java/net/dries007/tfc/client/screen/PetCommandScreen.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    @Override
+    protected void init()
+    {
+        super.init();
+        int y = 72;
+        for (TamableMammal.Command command : TamableMammal.Command.VALUES)
+        {
+            if (entity.willListenTo(command, true))
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/client/screen/PetCommandScreen.java:45]: This check needs to be moved, or duplicated, to the server, i.e. the packet on receive needs to check that this is true.

File: src/main/java/net/dries007/tfc/client/screen/PetCommandScreen.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            {
+                addRenderableWidget(new Button(width / 2 - 100, height / 4 + y, 200, 20, Helpers.translateEnum(command), b -> {
+                    PacketHandler.send(PacketDistributor.SERVER.noArg(), new PetCommandPacket(entity, command));
+                    Minecraft.getInstance().setScreen(null);
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/client/screen/PetCommandScreen.java:49]: I believe you also need a `playerInventory.player.containerMenu = playerInventory.player.inventoryMenu;` here. It probably doesn't matter, as I don't think this container has slots, but it still "closes" the container on client.

File: src/main/java/net/dries007/tfc/common/blocks/devices/QuernBlock.java [hunk 5]
```diff
@@ -97,9 +105,34 @@ private static InteractionResult insertOrExtract(Level level, QuernBlockEntity q
         return InteractionResult.sidedSuccess(level.isClientSide);
     }
 
+    public static void updateHandstone(Level level, BlockPos pos, QuernBlockEntity quern)
+    {
+
+    }
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/common/blocks/devices/QuernBlock.java]: ??

File: src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    {
+        return new Vec3(mob.getX(), getSurfaceYRespectingFluid(this, mob, level), mob.getZ());
+    }
+
+    public static class TFCWalkNodeEvaluator extends WalkNodeEvaluator
+    {
+        @Override
+        protected BlockPathTypes evaluateBlockPathType(BlockGetter level, boolean openDoors, boolean enterDoors, BlockPos pos, BlockPathTypes pathType)
+        {
+            // interpret leaves as open space
```
Comment (alcatrazEscapee) [inline src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java:71]: Should we be setting this in the leaves block? i.e. the `getBlockPathType` method or whatnot? It seems weird to be special casing it here.

**Review by alcatrazEscapee — CHANGES_REQUESTED** [review]

File: resources/data.py (hunk 2 above)
Comment (eerussianguy) [reply resources/data.py:393]: this tag is duplicated on line 169

File: src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java (hunk 6 above)
Comment (eerussianguy) [reply src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java:71]: generally we want some things to recognize leaves as leaves (aka something blocked), for example if another mod added a bird mob that lands on trees.

Comment (eerussianguy) [reply src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java:71]: this is also the vanilla pattern (ie, ravagers)

Comment (alcatrazEscapee) [reply src/main/java/net/dries007/tfc/common/entities/ai/TFCGroundPathNavigation.java:71]: in vanilla leaves are solid tho; and ravagers straight out break leaves. I don't think that's comparable, no?

**Pull request merged**